virtual_ecosystem/models/animal/animal_model.py

```diff
@@ -1076,9 +1181,96 @@ def metamorphose_community(self) -> None:
                 and (cohort.mass_current >= cohort.functional_group.adult_mass)
             ):
                 self.metamorphose(cohort)
+
+    def update_migrated_and_aquatic(self, dt: timedelta64) -> None:
+        """Handles updating timing on frozen migrated and aquatic cohorts.
+
+        Args:
+            dt: The amount of time passed in the update (days).
+
+        """
+
+        dt_float = float(dt / timedelta64(1, "D"))
+        for cohort in list(self.migrated_cohorts.values()):
+            cohort.remaining_time_away -= dt_float
+            if cohort.remaining_time_away <= 0:
+                self.reintegrate_cohort(cohort, source="migrated")
+
+        for cohort in list(self.aquatic_cohorts.values()):
+            cohort.remaining_time_away -= dt_float
+            if cohort.remaining_time_away <= 0:
+                self.reintegrate_cohort(cohort, source="aquatic")
+
+    def reintegrate_cohort(self, cohort: AnimalCohort, source: str) -> None:
+        """Handles integration of cohorts from migrated/aquatic to active status.
+
+        Args:
+            cohort: The animal cohort changing to active status.
+            source: Whether the cohort was migrated or aquatic.
+
+        """
+        if source == "migrated":
+            mortality_rate = cohort.constants.migration_mortality
+            self.migrated_cohorts.pop(cohort.id)
+        elif source == "aquatic":
+            mortality_rate = cohort.constants.aquatic_mortality
+            self.aquatic_cohorts.pop(cohort.id)
+
+        deaths = int(cohort.individuals * mortality_rate)
+        cohort.individuals -= deaths
+
+        if cohort.individuals > 0:
+            cohort.location_status = "active"
+            self.active_cohorts[cohort.id] = cohort
+
+            # Reintroduce cohort to its communities
+            self.update_community_occupancy(cohort, cohort.centroid_key)
+
+        else:
+            cohort.is_alive = False
+
+    def migrate_external(self, cohort: AnimalCohort) -> None:
+        """Handles the initiation of external migration events.
+
+        Args:
+            cohort: The migrating cohort.
+        """
+        # Remove cohort from community occupancy
+        self.abandon_communities(cohort)
+
+        # Move cohort to migration pool
+        cohort.location_status = "migrated"
+        cohort.remaining_time_away = cohort.constants.migration_residence_time
+        self.migrated_cohorts[cohort.id] = cohort
+        self.active_cohorts.pop(cohort.id)
+
+    def migrate_external_community(self) -> None:
+        """Cycles through all active cohorts and checks for external migration.
+
+        Only calls `trigger_external_migration` for cohorts that are seasonal migrators.
+        """
+        for cohort in list(self.active_cohorts.values()):
+            if (
+                cohort.functional_group.migration_type == "seasonal"
+                and cohort.is_migration_season()
+            ):
+                self.migrate_external(cohort)
```

Comment:
Are migration seasons a relevant concept if the migration type isn't seasonal? If not, I'd be inclined to make `is_migration_season` just return false in this case and document this behaviour in the docstring.